### PR TITLE
Allow ViewFeature tests to run in dnxcore50

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/DefaultViewComponentDescriptorProviderTest.cs [hunk 1]
```diff
@@ -81,7 +81,10 @@ protected override IEnumerable<TypeInfo> GetCandidateTypes()
                     GetAssemblyProvider()
                     .CandidateAssemblies
                     .SelectMany(a => a.DefinedTypes)
-                    .Select(t => t.GetTypeInfo());
+#if DNX451
+                    .Select(t => t.GetTypeInfo())
+#endif
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/DefaultViewComponentDescriptorProviderTest.cs]: Why not do this unconditionally? Where necessary could add
```
#if DNXCORE50
using System.Reflection;
#endif
```

**Comment by dougbu** [comment]
Looks good. What are the test counts when run with `dnxcore50` and `dnx451`?

**Comment by pranavkm** [comment]
DNX451 1732
DNXCore50: 238

There's probably a lot more that can be made to run in Dnx451 but I just wanted to get some test coverage for the view component helper test for now.

File: test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/DefaultViewComponentDescriptorProviderTest.cs (hunk 1 above)
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/DefaultViewComponentDescriptorProviderTest.cs]: `DefinedTypes` returns `TypeInfo` in DNXCore50 and `Type` in DNX451. It's kinda weird they broke this API.

**Comment by rynowak** [comment]
Is this checked in now?

File: test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/DefaultViewComponentDescriptorProviderTest.cs (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewComponents/DefaultViewComponentDescriptorProviderTest.cs]: ick but :ok:

**Comment by dougbu** [comment]
:shipit:

**Pull request closed without merging**